runner/lib/helpers/buffer-line-transform.js

```diff
@@ -0,0 +1,95 @@
+/* global Buffer */
+/* eslint-disable no-underscore-dangle,no-nested-ternary,no-plusplus */
+
+import { Transform } from 'stream';
+
+export default class BufferLineTransform extends Transform {
+  /**
+   * The BufferLineTransform is reading String or Buffer content from a Readable stream
+   * and writing each line as a Buffer in object mode
+   *
+   * @param {import('./buffer-line-transform.js').BufferLineTransformOptions} [options]
+   */
+  constructor(options) {
+    const { break: breakValue, breakEncoding, ...transformOptions } =
+      options || {};
+    super({ ...transformOptions, readableObjectMode: true });
+    this._breakValue = breakValue || 10;
+    this._breakEncoding = breakEncoding;
+
+    /** @type {number} */
+    let breakLength;
+    if (!breakValue || typeof breakValue === 'number') {
+      breakLength = 1;
+    } else if (Buffer.isBuffer(breakValue)) {
+      breakLength = breakValue.length;
+    } else {
+      breakLength = Buffer.from(breakValue, breakEncoding).length;
+    }
+    this._breakLength = breakLength;
+
+    /** @type {Array<Buffer>} */
+    this._chunks = [];
+  }
+
+  /**
+   * @param {any} chunk
+   * @param {BufferEncoding | 'buffer'} encoding
+   * @param {import('stream').TransformCallback} cb
+   * */
+  _transform(chunk, encoding, cb) {
+    try {
+      /** @type {Buffer} */
+      let buf =
+        Buffer.isBuffer(chunk) || encoding === 'buffer'
+          ? chunk
+          : Buffer.from(chunk, encoding);
+
+      while (buf.length) {
+        const offset = buf.indexOf(this._breakValue, 0, this._breakEncoding);
+
+        /** @type {number} */
+        let endOffset;
+        if (offset >= 0) {
+          endOffset = offset + this._breakLength;
+          if (this._chunks.length) {
+            const concatLength = this._chunks.reduce(
+              (acc, { length }) => acc + length,
+              endOffset,
+            );
```

Comment:
I can read this. A utility function might be nicer. A for loop might be kinder on both people and performance.

Reply:
I don't like `reduce` much either, but gotta admit that accumulation of a value is basically what it's originally meant for.